copycat/runtime/src/main/java/org/apache/kafka/copycat/runtime/rest/RestServer.java

```diff
@@ -0,0 +1,228 @@
+        context.setContextPath("/");
+        context.addServlet(servletHolder, "/*");
+
+        RequestLogHandler requestLogHandler = new RequestLogHandler();
+        Slf4jRequestLog requestLog = new Slf4jRequestLog();
+        requestLog.setLoggerName(RestServer.class.getCanonicalName());
+        requestLog.setLogLatency(true);
+        requestLogHandler.setRequestLog(requestLog);
+
+        HandlerCollection handlers = new HandlerCollection();
+        handlers.setHandlers(new Handler[]{context, new DefaultHandler(), requestLogHandler});
+
+        /* Needed for graceful shutdown as per `setStopTimeout` documentation */
+        StatisticsHandler statsHandler = new StatisticsHandler();
+        statsHandler.setHandler(handlers);
+        jettyServer.setHandler(statsHandler);
+        jettyServer.setStopTimeout(GRACEFUL_SHUTDOWN_TIMEOUT_MS);
+        jettyServer.setStopAtShutdown(true);
+
+        try {
+            jettyServer.start();
+        } catch (Exception e) {
+            throw new CopycatException("Unable to start REST server", e);
+        }
+
+        log.info("REST server listening at " + jettyServer.getURI() + ", advertising URL " + advertisedUrl());
+    }
+
+    public void stop() {
+        try {
+            jettyServer.stop();
+            jettyServer.join();
+        } catch (Exception e) {
+            throw new CopycatException("Unable to stop REST server", e);
+        } finally {
+            jettyServer.destroy();
+        }
+    }
+
+    /**
+     * Get the URL to advertise to other workers and clients. This uses the default connector from the embedded Jetty
+     * server, unless overrides for advertised hostname and/or port are provided via configs.
+     */
+    public String advertisedUrl() {
+        UriBuilder builder = UriBuilder.fromUri(jettyServer.getURI());
+        String advertisedHostname = config.getString(WorkerConfig.REST_ADVERTISED_HOST_NAME_CONFIG);
+        if (advertisedHostname != null && !advertisedHostname.isEmpty())
+            builder.host(advertisedHostname);
+        Integer advertisedPort = config.getInt(WorkerConfig.REST_ADVERTISED_PORT_CONFIG);
+        if (advertisedPort != null)
+            builder.port(advertisedPort);
+        else
+            builder.port(config.getInt(WorkerConfig.REST_PORT_CONFIG));
+        return builder.build().toString();
+    }
+
+
+    /**
+     * @param url               HTTP connection will be established with this url.
+     * @param method            HTTP method ("GET", "POST", "PUT", etc.)
+     * @param requestBodyData   Object to serialize as JSON and send in the request body.
+     * @param responseFormat    Expected format of the response to the HTTP request.
+     * @param <T>               The type of the deserialized response to the HTTP request.
+     * @return The deserialized response to the HTTP request, or null if no data is expected.
+     */
+    public static <T> T httpRequest(String url, String method, Object requestBodyData,
```

Comment:
What does this do? It looks like it is forwarding HTTP requests. Is it forwarding HTTP requests to the leader? I.e. all of the nodes accept HTTP requests, but non-leaders forward the request to the leader, and then relay back the response?